cogs/games.py

```diff
@@ -20,18 +20,203 @@
 # discord-py requirements
 import discord
 from discord.ext import commands
+from discord import Embed
 
 # Other utilities
 import re
+import os
+from time import time
+from pickle import load
+from random import choice, randint
+from typing import Dict, Tuple
 from .utils.dice_roll import dice_roll
 from .utils.clamp_default import clamp_default
+from .utils.hangman import HANG_LIST
 
 ROLL_PATTERN = re.compile(r'^(\d*)d(\d*)([+-]?\d*)$')
+NEWLINE = "\n"
 
 
 class Games(commands.Cog):
-    def __init__(self, bot):
+    def __init__(self, bot, hangman_tbl_name: str):
         self.bot = bot
+        my_path = os.path.abspath(os.path.dirname(__file__))
+        hangman_path = os.path.join(my_path, f"utils/{hangman_tbl_name}")
+        with open(hangman_path, "rb") as hangman_pkl:
+            self.hangman_dict: Dict[str, Tuple[str, str]] = load(hangman_pkl)
+
+    @commands.max_concurrency(1, per=commands.BucketType.channel, wait=False)
+    @commands.command(aliases=["hm"])
+    async def hangman(self, ctx, *, command: str = None):
+        """Play a nice game of hangman with internet strangers!
+        Guesses must be single letters (guesses must be lower case letters to be valid) or the entire correct word
+		Can either be called with "?{hm|hangman}" or "?{hm|hangman} x", where x is a valid category command
+        Get all categories/help by typing "?{hm|hangman} help"
+        """
+        if command == "help":
+            await ctx.send(
+                f"rules: 6 wrong guesses are allowed, guesses must be either the entire correct word or a single lowercase letter\nhere is a list of valid category commands: {sorted(self.hangman_dict.keys())}"
+            )
+            return
+        if command is None:
+            category = choice(list(self.hangman_dict.keys()))
+        else:
+            category = command
+        try:
+            word_list, cat_name = self.hangman_dict[category]
+            word = choice(word_list).lower()
+        except KeyError:
+            await ctx.send(
+                f"invalid category command, here is a list of valid commands: {sorted(self.hangman_dict.keys())}"
+            )
+            return
+        num_mistakes = 0
+        not_guessed = set(re.sub(r"[^a-z]", "", word))
+        incorrect_guesses = set()
+        first_line = "".join(char + " " if char not in not_guessed else "_ "
+                             for char in word).rstrip()
```

Comment:
instead of adding spaces here + rstrip, you could just join by space
```python
first_line = " ".join(char if char not in not_guessed else "_" for char in word)
```

Reply:
that rstrip call bothered me so much, thats a really good golf, i knew i was missing something
thx! (changed)